ras_party/controllers/account_controller.py

```diff
@@ -746,6 +746,15 @@ def put_email_verification(token, tran, session):
         # We set the user as verified on the OAuth2 server.
         set_user_verified(email_address)
 
+    if not respondent.pending_email_address and respondent.password_verification_token:
+        if not respondent.password_verification_token:
+            logger.info("Verification token not found")
+            raise NotFound("Verification token not found")
+
+        # Reset password token fields in the database since they are linked to the old email
+        delete_respondent_password_verification_token(respondent.party_uuid, session)
```

Comment:
I do wonder if these 2 are intrinsically linked have a parent function, maybe worth check other usages in the code

Reply:
They are linked but there's no parent function and adding it would require work to frontstage as well and I think is out of the scope of the card